samples/javascript/index.js

```diff
@@ -0,0 +1,253 @@
+const EventSourcePolyfill = require('eventsource');
+
+// The address of your Gradle Enterprise server
+const GRADLE_ENTERPRISE_SERVER_URL = process.argv.slice(2);
+
+// Authorization credentials
+const EXPORT_API_USER = process.env.EXPORT_API_USER;
+const EXPORT_API_PASSWORD = process.env.EXPORT_API_PASSWORD;
+const BASIC_AUTH_TOKEN = Buffer.from(`${EXPORT_API_USER}:${EXPORT_API_PASSWORD}`).toString('base64')
+
+// The point in time from which builds should be processed.
+// Values can be 'now', or a number of milliseconds since the UNIX epoch.
+// The time is the point in time that the build was published to the server.
+const PROCESS_FROM = 'now';
+
+// How many builds to process at one time.
+// If running with very fast network connection to the server,
+// this number can be increased for better throughput.
+const MAX_CONCURRENT_BUILDS_TO_PROCESS = 6;
+
+// A build event handler that calculates and logs the build duration.
+//
+// Each "on" method is an event handler that receives each instance of that type of event.
+// Please see https://docs.gradle.com/enterprise/export-api for more information about the event types.
+//
+// After all events have been received, the "complete()" method will be called if it exists.
+class BuildDurationEventsHandler {
+    constructor(build) {
+        this.buildId = build.buildId;
+    }
+
+    onBuildStarted(eventPayload) {
+        this.startTime = eventPayload.timestamp;
+    }
+
+    onBuildFinished(eventPayload) {
+        const endTime = eventPayload.timestamp;
+        console.log(`Build ${GRADLE_ENTERPRISE_SERVER_URL}/s/${this.buildId} completed in ${endTime - this.startTime}ms`);
+    }
+}
+
+// A build event handler that counts how many tasks of the build were cacheable.
+class CacheableTaskCountHandler {
+    constructor(build) {
+        this.buildId = build.buildId;
+        this.cacheableTaskCount = 0;
+    }
+
+    onTaskFinished(eventPayload) {
+        if (eventPayload.data.cacheable) {
+            this.cacheableTaskCount++;
+        }
+    }
+
+    complete() {
```

Comment:
Can this be `onCompleted()` instead? If the participle tense does not align with `oncomplete`, should we instead use `onTaskFinish()` and the like for consistency?

Reply:
methods with onXXX will be included as event handlers by `getHandledEventTypesForHandlerClass` I guess that's why the original creator decided to go without on